### PR TITLE
PMM 7182 create rule without channel filter

### PR DESCRIPTION
**Pull request by palash25**
[PMM-7182](https://jira.percona.com/browse/PMM-7182)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1367

- [ ] Links to other linked pull requests (optional). https://github.com/Percona-Lab/pmm-api-tests/pull/166
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: services/management/ia/rules_service.go [hunk 1]
```diff
@@ -321,8 +321,12 @@ func (s *RulesService) CreateAlertRule(ctx context.Context, req *iav1beta1.Creat
 		For:          req.For.AsDuration(),
 		Severity:     models.Severity(req.Severity),
 		CustomLabels: req.CustomLabels,
-		ChannelIDs:   req.ChannelIds,
 	}
+
+	if len(req.ChannelIds) > 0 {
+		params.ChannelIDs = req.ChannelIds
+	}
```
Comment (artemgavrilov) [inline services/management/ia/rules_service.go:328]: Why we need this condition? We have the same check in `models.CreateRule`

**Review by artemgavrilov — CHANGES_REQUESTED** [review]

File: models/database.go [hunk 2]
```diff
@@ -405,7 +405,7 @@ var databaseSchema = [][]string{
 			severity VARCHAR NOT NULL,
 			custom_labels TEXT,
 			filters JSONB,
-			channel_ids JSONB NOT NULL,
+			channel_ids JSONB,
```
[the remark this answers is not in the document]
Comment (palash25) [reply models/database.go]: added a new migration

File: services/management/ia/rules_service.go (hunk 1 above)
Comment (palash25) [reply services/management/ia/rules_service.go]: removed redundant condition

**Review by artemgavrilov — APPROVED** [review]

File: models/rule_helpers_test.go [hunk 3]
```diff
@@ -70,6 +70,36 @@ func TestRules(t *testing.T) {
...
+			require.NoError(t, err)
+			defer func() {
+				require.NoError(t, tx.Rollback())
+			}()
+
+			q := tx.Querier
+
+			templateName := createTemplate(t, q)
+
+			params := createCreateRuleParams(templateName, "", false)
```
Comment (BupycHuk) [inline models/rule_helpers_test.go:84]: the same comment here related to using `true/false` in function calls

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: models/rule_helpers_test.go (hunk 3 above)
Comment (palash25) [reply models/rule_helpers_test.go]: replaced it with this https://github.com/percona/pmm-managed/pull/609/files#diff-7df81ba599594e0fb11c381bb999cc71b71361ba3a9d852784f04e67c37b223cR38

File: models/rule_helpers_test.go [hunk 4]
```diff
@@ -221,8 +254,8 @@ func TestRules(t *testing.T) {
 	})
 }
 
-func createCreateRuleParams(templateName, channelID string) *models.CreateRuleParams {
-	return &models.CreateRuleParams{
+func createCreateRuleParams(templateName, channelID string, filter models.Filter) *models.CreateRuleParams {
```
Comment (BupycHuk) [inline models/rule_helpers_test.go:257]: we can replace filter with filters here, in that case we can remove `emptyFilter` and if on 276 line.

**Review by BupycHuk — APPROVED** [review]
LGTM, just one comment which simplify code

File: models/rule_helpers_test.go (hunk 4 above)
Comment (palash25) [reply models/rule_helpers_test.go]: done

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**